cubedash/summary/_stores.py

```diff
@@ -1123,21 +1121,26 @@ def search_items(
             offset
         )
 
-        for r in self.e_index.execute_query(query):
-            yield DatasetItem(
-                dataset_id=r.id,
-                bbox=_box2d_to_bbox(r.bbox) if r.bbox else None,
-                product_name=r.product_name,
-                geometry=(
-                    _get_shape(r.geometry, self.e_index.get_srid_name(r.geometry.srid))
-                    if r.geometry is not None
-                    else None
-                ),
-                region_code=r.region_code,
-                creation_time=r.creation_time,
-                center_time=r.center_time,
-                odc_dataset=(self.index.datasets.get(r.id) if full_dataset else None),
-            )
+        with self.e_index.execute_query_lazy(query) as result:
+            for r in result:
+                yield DatasetItem(
+                    dataset_id=r.id,
+                    bbox=_box2d_to_bbox(r.bbox) if r.bbox else None,
+                    product_name=r.product_name,
+                    geometry=(
+                        _get_shape(
+                            r.geometry, self.e_index.get_srid_name(r.geometry.srid)
+                        )
+                        if r.geometry is not None
+                        else None
+                    ),
+                    region_code=r.region_code,
+                    creation_time=r.creation_time,
+                    center_time=r.center_time,
+                    odc_dataset=(
```

Comment:
Since your name ends up in the git blame, you might be able to fit the parameter in a single line if you remove the parenthesis (not sure why the parenthesis was there in the first place).

You could also try taking the RHS of `geometry=` and putting that on a single line without its parenthesis and then `ruff format`, but I'm guessing Ruff will re-wrap that because the line is too long.